app/views/catalog/ds.html.erb

```diff
@@ -22,9 +21,7 @@
         # If the user is a repo admin or is permitted to manage both content and
         # rights, display raw xml editing
     %>
-      <%= if can?(:manage_item, @obj) && Settings.editable_datastreams.include?(params[:dsid])
-            link_to "Edit #{params[:dsid]}", ds_solr_document_path(params[:id], params[:dsid], edit: 'true'), title: params[:dsid], data: { blacklight_modal: 'preserve' }
-        end %>
+      <%= render 'edit_datastream_xml' if can?(:manage_item, @obj) && Settings.editable_datastreams.include?(params[:dsid]) %>
```

Comment:
I think a partial here is giving a lot of unnecessary indirection?  Why not just do the link_to here?  Isn't this easier to read?
```suggestion
      <% if can?(:manage_item, @obj) && Settings.editable_datastreams.include?(params[:dsid]) %>
        <%= link_to "Edit #{params[:dsid]}", ds_solr_document_path(params[:id], params[:dsid], edit: 'true'), title: params[:dsid], data: { blacklight_modal: 'preserve' } %>
      <% end %>
```

Reply:
Oh, I see, I misunderstood your earlier comment to mean I should extract that into a partial, not just a plan erb block... Whoops.